src/main/java/org/fcrepo/client/HistoricMementoBuilder.java

```diff
@@ -45,4 +47,9 @@ public HistoricMementoBuilder(final URI uri, final FcrepoClient client, final St
         UTC_RFC_1123_FORMATTER.parse(mementoDatetime);
         request.setHeader(MEMENTO_DATETIME, mementoDatetime);
     }
+
+    @Override
+    public HistoricMementoBuilder addTransaction(final TransactionURI transaction) {
```

Comment:
@bbpennel The only tricky piece with this is the `HistoricMementoBuilder` since it inherits from the `PostBuilder`. I can shuffle things around so that uses the `BodyRequestBuilder` instead along with the two methods from the `PostBuilder` in order to keep parity with its old behavior. Then I can add the `addTransaction` method to other Builders which use them. 

I almost wonder if an additional interface is really needed since we're overriding `addTransaction` in every class to get the correct type in the return. But as long as the above sounds good I can have changes up shortly.

Reply:
It seems like in either approach, having `HistoricMementoBuilder` inherit from `PostBuilder` would introduce an unwanted `addTransaction` method, so switching over to inheriting from `BodyRequestBuilder` makes sense.

And you're right, there doesn't need to be an `interface`, and it doesn't look like we're using interfaces in this project at all actually. We can just change RequestBuilder.addTransaction to being protected and then leave it up to the concrete builders to make it public I think.